packages/side-nav/src/SideNavGroup.tsx

```diff
@@ -1,67 +1,80 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
 import { Transition } from 'react-transition-group';
-import ChevronRightIcon from '@leafygreen-ui/icon/dist/ChevronRight';
-import { css, cx } from '@leafygreen-ui/emotion';
-import { uiColors } from '@leafygreen-ui/palette';
-import { createDataProp, OneOf } from '@leafygreen-ui/lib';
+import {
+  createDataProp,
+  IdAllocator,
+  OneOf,
+  isComponentType,
+} from '@leafygreen-ui/lib';
 import { useUsingKeyboardContext } from '@leafygreen-ui/leafygreen-provider';
+import ChevronRight from '@leafygreen-ui/icon/dist/ChevronRight';
+import { prefersReducedMotion } from '@leafygreen-ui/a11y';
+import { uiColors } from '@leafygreen-ui/palette';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { spacing } from '@leafygreen-ui/tokens';
+import CollapsedSideNavItem from './CollapsedSideNavItem';
 import {
   ulStyleOverrides,
   sideNavItemSidePadding,
   sideNavWidth,
 } from './styles';
 
+const menuGroupIdAllocator = IdAllocator.create('side-nav-group');
```

Comment:
could also change this var name just for consistency

Reply:
Done